src/tribol/physics/Mortar.hpp

```diff
@@ -159,6 +159,80 @@ void ComputeConstraintJacobian<SINGLE_MORTAR, DUAL>( SurfaceContactElem& elem );
  */
 void ComputeSingleMortarJacobian( SurfaceContactElem& elem );
 
+#ifdef TRIBOL_USE_ENZYME
+/*!
+ * \brief Computes frictionless mortar forces and Jacobians (using Enzyme) following Puso and Laursen (2004)
+ *
+ * \param [in] cs pointer to the coupling scheme
+ *
+ * \return 0 if no error
+ *
+ */
+int ApplyNormalEnzyme( CouplingScheme* cs );
+
+/**
+ * @brief Computes the frictionless mortar forces for a 3D quad element (following Puso and Laursen (2004))
+ *
+ * @param [in] x1 Nodal coordinates for element 1 (stored by node, e.g. [x0, x1, x2, x3, y0, y1, y2, y3, z0, z1, z2,
+ * z3])
+ * @param [in] n1 Nodal unit normal vectors for element 1 (stored by node, e.g. [x0, x1, x2, x3, y0, y1, y2, y3, z0, z1,
+ * z2, z3])
+ * @param [in] p1 Nodal pressures for element 1
+ * @param [out] f1 Nodal forces for element 1 (stored by node, e.g. [x0, x1, x2, x3, y0, y1, y2, y3, z0, z1, z2, z3])
+ * @param [out] g1 Nodal gaps for element 1
+ * @param [in] size1 Number of nodes on element 1
+ * @param [in] x2 Nodal coordinates for element 2 (stored by node, e.g. [x0, x1, x2, x3, y0, y1, y2, y3, z0, z1, z2,
+ * z3])
+ * @param [out] f2 Nodal forces for element 2 (stored by node, e.g. [x0, x1, x2, x3, y0, y1, y2, y3, z0, z1, z2, z3])
+ * @param [in] size2 Number of nodes on element 2
+ */
+void ComputeMortarForceEnzyme( const RealT* x1, const RealT* n1, const RealT* p1, RealT* f1, RealT* g1, int size1,
+                               const RealT* x2, RealT* f2, int size2 );
+
+/**
+ * @brief Computes the frictionless mortar forces and Jacobian for a 3D quad element (following Puso and Laursen (2004))
```

Comment:
This is really hard to read. Perhaps this can be reformatted with clang format off just to have some alignment.

Reply:
Added some indentation and clang-format didn't complain